autotest/ogr/ogr_mssqlspatial.py

```diff
@@ -559,6 +559,43 @@ def test_ogr_mssqlspatial_datatypes():
 #
 
 
+def test_ogr_mssqlspatial_bulk_insert():
+    """Test issue GH https://github.com/OSGeo/gdal/issues/7787"""
+
+    if gdaltest.mssqlspatial_ds is None:
+        pytest.skip()
+
+    use_bcp = "BCP" in gdal.GetDriverByName("MSSQLSpatial").GetMetadataItem(
+        gdal.DMD_LONGNAME
+    )
+
+    if not use_bcp:
+        pytest.skip()
+
+    gdal.SetConfigOption("MSSQLSPATIAL_BCP_SIZE", "2")
```

Comment:
using gdal.SetConfigOption() is quite dangerous as it must be done, and in all code paths (regular & exceptions)
It is better to use the gdaltest.config_option() context manager (and indenting properly what code must be in that scope)
```suggestion
    with gdaltest.config_option("MSSQLSPATIAL_BCP_SIZE", "2"):
```